File: examples/Java/hwserver.java
//
//  Hello World server in Java
//  Binds REP socket to tcp://*:5555
//  Expects "Hello" from client, replies with "World"
//

import org.zeromq.ZMQ;

public class hwserver{

    public static void main (String[] args) throws Exception{
        ZMQ.Context context = ZMQ.context(1);
        //  Socket to talk to clients
        ZMQ.Socket socket = context.socket(ZMQ.REP);
        socket.bind ("tcp://*:5555");

        while (!Thread.currentThread ().isInterrupted ()) {
            byte[] reply = socket.recv(0);
            System.out.println("Received Hello");
            String request = "World" ;
            socket.send(request.getBytes (), 0);
            Thread.sleep(1000); //  Do some 'work'
        }
        socket.close();
        context.term();
    }
}

File: examples/Clojure/hwserver.clj
;; Hello World server in Clojure
;; Binds REP socket to tcp://*:5555
;; Expects "Hello" from client, replies with "World"

(ns hwserver
  (:require [zeromq.zmq :as zmq]))

(defn -main []
  (let [context (zmq/context 1)]
    (with-open [socket (doto (zmq/socket context :rep)
                         (zmq/bind "tcp://*:5555"))]
      (while (not (.. Thread currentThread isInterrupted))
        (let [reply (zmq/receive socket)]
          (println "Received Hello")
          ;; Do some work
          (Thread/sleep 1000)
          (zmq/send-str socket "World"))))))
